integration-tests/actions/seth/refund.go

```diff
@@ -262,94 +262,108 @@ func ReturnFunds(log zerolog.Logger, sethClient *seth.Client, chainlinkNodes []c
-				GasFeeCap:  estimations.GasFeeCap,
-				GasTipCap:  estimations.GasTipCap,
-				TxTimeout:  &txTimeout,
-			}
+	if sethClient.Cfg.IsExperimentEnabled(seth.Experiment_SlowFundsReturn) {
+		txPriority = "slow"
+		thirtyMinutes := time.Duration(30 * time.Minute)
+		txTimeout = thirtyMinutes
+	}
 
-			_, err = SendFunds(log, sethClient, payload)
-			if err != nil {
-				handler := OvershotTransferRetrier{maxRetries: 10, nextRetrier: &InsufficientFundTransferRetrier{maxRetries: 10, nextRetrier: &GasTooLowTransferRetrier{maxGasLimit: sethClient.Cfg.Network.TransferGasFee * 10}}}
-				err = handler.Retry(context.Background(), log, sethClient, err, payload, 0)
-				if err != nil {
-					log.Error().
-						Err(err).
-						Str("Address", fromAddress.String()).
-						Msg("Failed to return funds from Chainlink node to default network wallet")
-					failedReturns = append(failedReturns, fromAddress)
-				}
-			}
-		}
+	estimations := sethClient.CalculateGasEstimations(seth.GasEstimationRequest{
+		GasEstimationEnabled: sethClient.Cfg.Network.GasPriceEstimationEnabled,
+		FallbackGasPrice:     sethClient.Cfg.Network.GasPrice,
+		FallbackGasFeeCap:    sethClient.Cfg.Network.GasFeeCap,
+		FallbackGasTipCap:    sethClient.Cfg.Network.GasTipCap,
+		Priority:             txPriority,
+	})
+
+	var maxTotalGasCost *big.Int
+	if sethClient.Cfg.Network.EIP1559DynamicFees {
+		maxTotalGasCost = new(big.Int).Mul(big.NewInt(0).SetInt64(sethClient.Cfg.Network.TransferGasFee), estimations.GasFeeCap)
+	} else {
+		maxTotalGasCost = new(big.Int).Mul(big.NewInt(0).SetInt64(sethClient.Cfg.Network.TransferGasFee), estimations.GasPrice)
 	}
 
-	if len(failedReturns) > 0 {
-		return fmt.Errorf("failed to return funds from Chainlink nodes to default network wallet for addresses: %v", failedReturns)
+	toSend := new(big.Int).Sub(balance, maxTotalGasCost)
+
+	if toSend.Cmp(big.NewInt(0)) <= 0 {
+		log.Warn().
+			Str("Address", fromAddress.String()).
+			Str("Estimated maximum total gas cost", maxTotalGasCost.String()).
+			Str("Balance", balance.String()).
+			Str("To send", toSend.String()).
+			Msg("Not enough balance to cover gas cost. Skipping return.")
+
+		return nil
 	}
 
-	log.Info().Msg("Successfully returned funds from all Chainlink nodes to default network wallets")
+	payload := FundsToSendPayload{
+		ToAddress:  sethClient.Addresses[0],
+		Amount:     toSend,
+		PrivateKey: fromPrivateKey,
+		GasLimit:   &sethClient.Cfg.Network.TransferGasFee,
+		GasPrice:   estimations.GasPrice,
+		GasFeeCap:  estimations.GasFeeCap,
+		GasTipCap:  estimations.GasTipCap,
+		TxTimeout:  &txTimeout,
+	}
+
+	_, err = SendFunds(log, sethClient, payload)
+	if err != nil {
+		handler := OvershotTransferRetrier{maxRetries: 10, nextRetrier: &InsufficientFundTransferRetrier{maxRetries: 10, nextRetrier: &GasTooLowTransferRetrier{maxGasLimit: sethClient.Cfg.Network.TransferGasFee * 10}}}
+		err = handler.Retry(context.Background(), log, sethClient, err, payload, 0)
```

Comment:
it should use the context provided from the test

Reply:
will fix